src/OpenLoco/src/Windows/ObjectSelectionWindow.cpp

```diff
@@ -618,6 +685,36 @@ namespace OpenLoco::Ui::Windows::ObjectSelectionWindow
         }
     }
 
+    static void drawSearchBox(Window& self, Gfx::RenderTarget* rt)
+    {
+        char* textBuffer = (char*)StringManager::getString(StringIds::buffer_2039);
+        strcpy(textBuffer, inputSession.buffer.c_str());
```

Comment:
buffer_2039 has a known max size. Could you use a safer strcpy

Reply:
```suggestion
        strncpy(textBuffer, inputSession.buffer.c_str(), 256);
```